frontend/models/product.ts

```diff
@@ -0,0 +1,159 @@
+import { IFIXIT_ORIGIN } from '@config/env';
+import { filterNullableItems, invariant } from '@helpers/application-helpers';
+import { isRecord } from '@ifixit/helpers';
+import {
+   ShopCredentials,
+   getShopifyStorefrontSdk,
+   MoneyV2,
+   CurrencyCode,
+} from '@lib/shopify-storefront-sdk';
+
+export async function findProduct(shop: ShopCredentials, handle: string) {
+   const storefront = getShopifyStorefrontSdk(shop);
+
+   const response = await storefront.findProduct({
+      handle,
+   });
+   if (response.product == null) {
+      return null;
+   }
+   const variants = response.product.variants.nodes.map((variant) => {
+      return {
+         ...variant,
+         formattedPrice: formatPrice(variant.priceV2),
+         formattedCompareAtPrice: variant.compareAtPriceV2
+            ? formatPrice(variant.compareAtPriceV2)
+            : null,
+         kitContents: variant.kitContents?.value ?? null,
+         note: variant.note?.value ?? null,
+         disclaimer: variant.disclaimer?.value ?? null,
+         warning: variant.warning?.value ?? null,
+         specifications: variant.specifications?.value ?? null,
+         warranty: variant.warranty?.value ?? null,
+      };
+   });
+   const variantSku = variants.find((variant) => variant.sku != null)?.sku;
+   if (variantSku == null) {
+      console.warn(`No sku found for product "${handle}"`);
+      return null;
+   }
+   const iFixitProductId = computeIFixitProductId(variantSku);
+   const reviewsData = await fetchProductReviews(
+      IFIXIT_ORIGIN,
+      iFixitProductId
+   );
+
+   return {
+      ...response.product,
+      iFixitProductId,
+      variants,
+      images: response.product.images.nodes.map((node) => node),
```

Comment:
Nope! It's probably because I've refactored from using `edges` to directly use `nodes` and I didn't realize that mapping was no longer necessary. Thanks for pointing this out 👍